README.md

```diff
@@ -88,6 +89,34 @@ julia> post = sample(DynamicHMCChain, m(x=x) | (y=y,))
 (σ = 1.0±0.15, β = 0.503±0.26, α = 10.2±0.25)
 ```
 
+## How is Soss different from [Turing](https://turing.ml/dev/)?
+
+- In Soss, models are first class, and can be composed or nested. At the time of this writing, it looks like nested models will be coming soon to Turing, see [this PR](https://github.com/TuringLang/DynamicPPL.jl/pull/233).
+- Soss has been updated to use [MesureTheory.jl](https://github.com/cscherrer/MeasureTheory.jl), though everything from Distributions.jl is still available.
+- Soss allows model transformations. This can be used, for example, to easily express predictive distributions or Markov blanket as a new model.
+- Most of the focus of Soss is at the syntactic level; inference works in terms of "primitives" that transform the model's abstract syntax tree (AST) to new code. As a result, there's complete flexibility for new applications. In particular the generated code is not required to use "Soss-specific" data structures in any way. 
```

Comment:
> In particular the generated code is not required to use "Soss-specific" data structures in any way.

I'm curious, what data structures do you have in mind? In general, you can use arbitrary Julia packages (such as SciML etc.) inside of Turing models. And you can customize the type of the resulting chain arbitrarily.

Reply:
That's from the user side, the point here is from the developer side. So I'm mostly thinking of things like `VarInfo`. In Soss, generated code can be an arbitrary function of the model AST, so you can do lots of very nonstandard things much more easily. So in addition to `rand` and `logdensity` primitives, we have things like
- `xform` to find a transform from R^n, 
- `testvalue` which is like `rand` but instead just calls `testvalue` at each point, 
- `basemeasure` that aggregates calls to `basemeasure` to end up with a base measure for the model as a whole
- `entropy` that computes the entropy of a model through a combination of particles and calls to `Distributions.entropy`

All of these are built-in, but there's nothing magic about them; end-users can add more primitives like this. More the core of it is that Soss doesn't dictate anything at all about how models can be evaluated.

Reply:
> More the core of it is that Soss doesn't dictate anything at all about how models can be evaluated.

That's not the case in Turing either. We define convenient functions to evaluate the log joint, log prior, or log likelihood but it's trivial to perform any other evaluation of the model (one could even use this to derive a symbolic representation of the model, similar to `modelingtoolkitize` in SciML, I think - I haven't tried it but I think it would be a cool project). We don't define `rand` (maybe we should) but you can achieve the same by just calling the model. `xform` seems to correspond to the Bijectors functionality in Turing? There are surely some things that work better in Soss due to its design but it seems many/most evaluations can be performed with Turing as well (possibly less efficient) and it can also be extended very easily (this is one of the main design principle - the model is "just" a regular Julia function, one should be able to use arbitrary Julia code in it, and it should be easily extendable).